core/shared/src/main/scala/zio/ZManaged.scala

```diff
@@ -2291,4 +2285,14 @@ object ZManaged extends ZManagedPlatformSpecific {
 
   private[zio] def succeedNow[A](r: A): ZManaged[Any, Nothing, A] =
     ZManaged(IO.succeedNow((Finalizer.noop, r)))
+
+  implicit final class RefineToOrDieOps[R, E <: Throwable, A](private val self: ZManaged[R, E, A]) extends AnyVal {
+
+    /**
+     * Keeps some of the errors, and terminates the fiber with the rest.
+     */
+    def refineToOrDie[E1 <: E: ClassTag](implicit ev: CanFail[E]): ZManaged[R, E1, A] =
```

Comment:
Can we keep it on the class with implicit evidence? (`implicit ev: E <:< Throwable`). Better discoverability that way.

Reply:
Yep, but the cost is that we can no longer constrain `E1 <: E` and the test cases won't pass (#2355). This PR is just a copy of what was done for `zio.ZIO` (#2360) to fix that.